narwhal/consensus/src/bullshark.rs

```diff
@@ -97,27 +97,31 @@ impl Bullshark {
                 let next_leader = (round/2 - 1) as usize % committee.size();
                 let authorities = committee.authorities().collect::<Vec<_>>();
 
-                authorities.get(next_leader).unwrap().id()
+                (*authorities.get(next_leader).unwrap()).clone()
             } else {
                 // Elect the leader in a stake-weighted choice seeded by the round
-                committee.leader(round).id()
+                committee.leader(round)
             }
         }
     }
 
-    /// Returns the certificate (and the certificate's digest) originated by the leader of the
-    /// specified round (if any).
+    /// Returns the certificate originated by the leader of the specified round (if any). The Authority
+    /// leader of the round is always returned and that's irrespective of whether the certificate exists
+    /// as that's deterministically determined.
     fn leader<'a>(
         committee: &Committee,
         round: Round,
         dag: &'a Dag,
-    ) -> Option<&'a (CertificateDigest, Certificate)> {
+    ) -> (Option<&'a Certificate>, Authority) {
```

Comment:
nit: I would reorder the elements, e.g. `(Authority, Option<&'a Certificate>)`, since this function is named `leader()`.